gateway-messages/src/sp_impl.rs

```diff
@@ -417,6 +420,15 @@ pub trait SpHandler {
     fn start_host_flash_hash(&mut self, slot: u16) -> Result<(), SpError>;
 
     fn get_host_flash_hash(&mut self, slot: u16) -> Result<[u8; 32], SpError>;
+
+    /// Unlocks the tech port if the challenge and response are compatible
+    fn unlock(
+        &mut self,
+        vid: Self::VLanId,
+        challenge: UnlockChallenge,
+        response: UnlockResponse,
+        time_sec: u32,
+    ) -> Result<(), MonorailError>;
```

Comment:
What's the advantage of moving this into the trait? Is this to avoid the extra monorail step?

Reply:
I have not worked in this code base before, so it may just be a mistake. But I was guided by the module-level doc-comment `//! Behavior implemented by both real and simulated SPs.` I figured that with oxidecomputer/omicron#8994, the trait would now be the right place for this operation.